modules/claims_api/lib/pdf_generator_service/pdf_client.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module ClaimsApi
+    ##
+    # Class to interact with the pdf generator service
+
+    class PDFClient
+
+        def initialize(request_body = nil)
+            @request_body = request_body
+        end
+
+        def generate_pdf
+            url = Settings.pdf_generator.url
+            path = Settings.pdf_generator.path
+            content_type = Settings.pdf_generator.content_type
+            conn = Faraday.new("#{url}#{path}",
+                        ssl: { verify: false },
+                        headers: { 'Content-Type': "#{content_type}" }) do |f|
+                f.request :json
+                f.response :raise_error
+                f.response :logger,
+                Rails.logger,
+                headers: true,
+                bodies: true,
+                log_level: :debug
+                f.adapter Faraday.default_adapter
+            end
+            response = conn.post('526', '{"data": {}}').body
+            File.open('526.pdf', 'wb') { |file| file.write(response) }
```

Comment:
I'd recommend [generating a tempfile](https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/common/file_helpers.rb#L15) and updating the response to set the filename for the user. While not terribly likely, using the same target file path & filename could lead to overriding a file that another user has generated, leading to the possibility of Veteran A getting Veteran B's file.

Reply:
Address this previously but neglected to respond. Very good point. Updated to use FileHelpers.generate_temp_file.